File: university-service/src/main/java/ua/com/foxminded/courseproject/exceptions/StudentConflictException.java
package ua.com.foxminded.courseproject.exceptions;

import ua.com.foxminded.courseproject.dto.StudentDto;

public class StudentConflictException extends RuntimeException {

    static private final String message = "Student %s %s %s already exists.";

    public StudentConflictException() {
        super();
    }

    public StudentConflictException(String s, Throwable cause) {
        super(s, cause);
    }

    public StudentConflictException(Throwable cause) {
        super(cause);
    }

    public StudentConflictException(String s) {
        super(s);
    }

    public StudentConflictException (StudentDto studentDto){
        super(message.formatted(studentDto.getFirstName(), studentDto.getLastName(), studentDto.getBirthDay().toString()));
    }

}

File: university-service/src/main/java/ua/com/foxminded/courseproject/exceptions/StudentConflictException.kt
package ua.com.foxminded.courseproject.exceptions

import ua.com.foxminded.courseproject.dto.StudentDto

class StudentConflictException : RuntimeException {
    constructor() : super()
    constructor(s: String, cause: Throwable) : super(s, cause)
    constructor(cause: Throwable) : super(cause)
    constructor(s: String) : super(s)
    constructor(studentDto: StudentDto) : super(
        Companion.message.format(
            studentDto.firstName,
            studentDto.lastName,
            studentDto.birthDay.toString()
        )
    )

    companion object {
        private const val message = "Student %s %s %s already exists."
    }
}
